Zend/tests/readonly_props/readonly_clone.phpt

```diff
@@ -0,0 +1,66 @@
+--TEST--
+clone can write to readonly properties
+--FILE--
+<?php
+
+class Counter
+{
+    private static int $counter = 0;
+
+    public readonly int $count;
+    private readonly int $foo;
+
+    public function __construct()
+    {
+        $this->count = ++self::$counter;
+        $this->foo = 0;
+    }
+
+    public function count(?int $count = null): static
+    {
+        $new = clone $this;
+        $new->count = $count ?? ++self::$counter;
+
+        return $new;
+    }
+
+    public function __clone()
+    {
+        if (is_a(debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2)[1]['class'] ?? '', self::class, true)) {
+            unset($this->count);
+        } else {
+            $this->count = ++self::$counter;
+        }
+        $this->foo = 1;
```

Comment:
For some reason this line makes tests fail and I can't figure out what's missing in the implementation.
Help wanted :pray: